module/os/freebsd/zfs/vdev_geom.c

```diff
@@ -920,7 +921,11 @@ vdev_geom_open(vdev_t *vd, uint64_t *psize, uint64_t *max_psize,
 	vd->vdev_nowritecache = B_FALSE;
 
 	/* Inform the ZIO pipeline that we are non-rotational */
-	vd->vdev_nonrot = (g_getattr("GEOM::rotation_rate", cp, &rate) == 0);
+        error = g_getattr("GEOM::rotation_rate", cp, &rate);
+        if (error == 0 && rate == 1)
+                vd->vdev_nonrot = B_TRUE;
+        else
+                vd->vdev_nonrot = B_FALSE;
```

Comment:
Thanks, LGTM! One small nit, the whitespace appears to be different than the surrounding code.

Reply:
I just checked the SSDs in my system and found geom (at least in userland) reports 0 as the rotation rate. I imagine this would be the same value found in the kernel, so I'm starting to question why FreeBSD is checking for the value 1. @amotin is that 1 a typo?

Reply:
Further investigation shows that `DISK_RR_NON_ROTATING` is defined as 1 in `geom_disk.h`. It would be best to use that constant here for clarity.

Reply:
And indeed GEOM does translate `DISK_RR_NON_ROTATING` into "0" for userland in `geom_disk.c`.